src/components/show-hide-when/display-when.ts

```diff
@@ -1,46 +1,134 @@
-import { NgZone } from '@angular/core';
+import { Input, NgZone } from '@angular/core';
 
+import { Config } from '../../config/config';
 import { Platform } from '../../platform/platform';
 
+const QUERY: { [key: string]: string }  = {
+  xs: '(min-width: 0px)',
+  sm: '(min-width: 576px)',
+  md: '(min-width: 768px)',
+  lg: '(min-width: 992px)',
+  xl: '(min-width: 1200px)'
+};
 
 /**
  * @hidden
  */
 export class DisplayWhen {
+  private _resizeObs: any;
+  private _query: string;
+  private _rmListener: any;
+
   isMatch: boolean = false;
   conditions: string[];
-  resizeObs: any;
 
-  constructor(conditions: string, public _plt: Platform, public zone: NgZone) {
+  @Input()
+  set when(query: string) {
+    if (typeof query === 'boolean') {
+      this.isMatch = this._query = query;
+    } else {
+      query = this._query = query.toLowerCase();
+      this.checkConditions(query);
+    }
+  }
+
+  get when(): string {
+    return this._query;
+  }
+
+  constructor(conditions: string, public _config: Config, public _plt: Platform, public zone: NgZone) {
+    // this.checkConditions(conditions);
+  }
+
+  checkConditions(conditions: string): void {
     if (!conditions) return;
 
-    this.conditions = conditions.replace(/\s/g, '').split(',');
+    // Split the conditions on commas, unless the comma is inside of parantheses
+    // because this means it is inside of a media query
+    this.conditions = conditions.replace(/\s/g, '').split(/\,(?![^\(]*\))/g);
```

Comment:
I would add change this line to:
```
this.conditions = conditions = conditions.replace(/\s/g, '').split(/\,(?![^\(]*\))/g);
```

then use `conditions` in the rest of the function. Remember that local variables are minified and have a better performance.